src/Diag/Result.hs

```diff
@@ -275,3 +277,16 @@ section name content =
 
 subsection :: Doc AnsiStyle -> [Doc AnsiStyle] -> Doc AnsiStyle
 subsection name = vsep . map (\single -> annotate (color Yellow) name <> line <> line <> indent 2 single <> line)
+
+-- | Log all encountered errors and warnings associated with 'Result a'
+--
+-- - On failure, the failure is logged with the provided @sevOnErr@ severity
+--
+-- - On success, the associated warnings are logged with the provided
+--   @sevOnSuccess@ severity
+flushLogs :: Has Logger sig m => Severity -> Severity -> Result a -> m ()
+flushLogs sevOnErr _ (Failure ws eg) = Effect.Logger.log sevOnErr (renderFailure ws eg)
+flushLogs _ sevOnSuccess (Success ws _) = do
+  case renderSuccess ws of
+    Nothing -> pure ()
+    Just rendered -> Effect.Logger.log sevOnSuccess rendered
```

Comment:
This can be defined in terms of `Control.Carrier.Diagnostics.withResult`:

```haskell
flushLogs s1 s2 res = withResult s1 s2 res (void . pure)
```

This should probably be removed (I don't think it belongs in this module anyway), and the `(void . pure)` should be inlined at the call site.

Reply:
Alternatively, you can define it alongside `withResult`, and define it in terms of `withResult` as well, so we don't duplicate the code.

Reply:
Addressed in https://github.com/fossas/fossa-cli/pull/790/commits/b558311435073a8500f63e5365ea3304ffc90d31